tests/test_validation.py

```diff
@@ -0,0 +1,2803 @@
+    with patch('sphinxval.utils.config.logpath', './tests/logs'):
+        if not os.path.exists(config.logpath):
+            os.mkdir(config.logpath)
+
+        config_file = pathlib.Path('tests/log_config_tests.json')
+        with open(config_file) as f_in:
+            config_ = json.load(f_in)
+        logging.config.dictConfig(config_)
+
+
+
+def utility_load_forecast(filename, energy_channel):
+    """
+    Utility function for unit testing. Loads forecast object directly from JSON.
+       Input:
+            :filename: (string) forecast JSON file.
+            :energy_channel: (dict) 
+       Output:
+            :forecast: resultant Forecast() object.
+    """
+    forecast_dict = vjson.read_in_json(filename)
+    forecast, _ = vjson.forecast_object_from_json(forecast_dict, energy_channel)
+    forecast.source = filename
+    return forecast
+
+def utility_load_observation(filename, energy_channel):
+    """
+    Utility function for unit testing. Loads observation object directly from JSON.
+       Input:
+            :filename: (string) observation JSON file.
+            :energy_channel: (dict) 
+       Output:
+            :observation: resultant Observation() object.
+    """
+    observation_dict = vjson.read_in_json(filename)
+    observation = vjson.observation_object_from_json(observation_dict, energy_channel)
+    observation.source = filename   
+    return observation
+
+
+def utility_match_sphinx(all_energy_channels, model_names, obs_objs, model_objs):
+
+    utility_setup_logging()
+
+    matched_sphinx, all_observed_thresholds, observed_sep_events =\
+        match.match_all_forecasts(all_energy_channels, model_names,
+            obs_objs, model_objs)
+
+    return matched_sphinx, all_observed_thresholds, observed_sep_events
+    
+def utility_get_verbosity():
+    """
+    Sets verbosity for unit test suite.
+    """
+    if ('-v' in sys.argv) or ('--verbose' in sys.argv):
+        verbosity = 2
+    elif ('--quiet' in sys.argv):
+        verbosity = 0
+    else:
+        verbosity = 1
+    return verbosity
+
+def utility_delete_output():
+    """
+    Function to call at the start of each unittest to delete the contents 
+    of the output folder 
+    """
+    shutil.rmtree('./tests/output')
+
+    validate.prepare_outdirs()
+
+def make_docstring_printable(function):
+    """
+    Decorator method @make_docstring_printable.
+    Allows function to call itself to access docstring information.
+    Used for verbose unit test outputs.
+    """
+    def wrapper(*args, **kwargs):
+        return function(function, *args, **kwargs)
+    return wrapper
+
+def attributes_of_sphinx_obj(keyword, sphinx_obj, energy_channel_key, threshold_key):
+    """
+    Function that takes the keyword from the SPHINX dataframe and matches the keyword
+    to the matched sphinx object to compare/assertEqual to - ensures that the 
+    dataframe is being correctly built
+    """
+    if keyword == "Model": 
```

Comment:
I suggest shortening/simplifying this by changing the massive if/else block into a set of dictionary searches organized per object level in `sphinx_obj`.  This can be done for all the "trivial" mappings, and then the more complicated mappings can be handled at the end.  Example:

```python
top_attrs = {
  'Mismatch Allowed' : 'mismatch',
 'Observed Start Time': 'observed_start_time',
  'Observed End Time': 'observed_end_time',
 ...
}

if keyword in top_attrs:
  return getattr(sphinx_obj, top_attrs[keyword], None)

prediction_attrs = {
  'Forecast Source' : 'source',
 'Forecast Path' : 'path',
...
}

if keyword in prediction_attrs:
  return getattr(sphinx_obj.prediction, top_attrs[keyword], None)
```

Then you can handle the more complex cases that need deeper if/thens in the way you have done here.